src/commands/src/shared/next-gen-stage.ts

```diff
@@ -0,0 +1,46 @@
+import { executeCliCommand, getCommitHash, getPatchId } from '../helpers';
+
+interface StageParams {
+  repoDir: string;
+  mutPrefix: string;
+  projectName: string;
+  bucket: string;
+  url: string;
+  patchId?: string;
+  commitBranch: string;
+}
+
+export async function nextGenStage({
+  repoDir,
+  mutPrefix,
+  projectName,
+  bucket,
+  url,
+  patchId,
+  commitBranch,
+}: StageParams) {
+  let hostedAtUrl = `${url}/${mutPrefix}/docsworker/${commitBranch}/`;
+  let prefix = mutPrefix;
+
+  const commandArgs = ['public', bucket, '--stage'];
+
+  if (patchId && projectName === mutPrefix) {
+    const [commitHash, patchId] = await Promise.all([getCommitHash(repoDir), getPatchId(repoDir)]);
```

Comment:
Question: Why do we get the commit hash and patch ID again? Is there a reason we don't pass in and use `patchId` and `commitHash` from the call to `prepareBuildAndGetDependencies` earlier?